src/runtime/mini-prompt-api.js

```diff
@@ -0,0 +1,240 @@
+    );
+
+    /** @private @const {!./client-event-manager.ClientEventManager} */
+    this.eventManager_ = deps.eventManager();
+  }
+
+  /**
+   * Does the setup required for the mini prompt's display, including injecting
+   * the mini prompt's css.
+   */
+  init() {
+    const head = this.doc_.getHead();
+    if (!head) {
+      return;
+    }
+
+    const url = '$assets$/swg-mini-prompt.css';
+    const existing = head.querySelector(`link[href="${url}"]`);
+    if (existing) {
+      return;
+    }
+
+    // <link rel="stylesheet" href="..." type="text/css">
+    head.appendChild(
+      createElement(this.doc_.getWin().document, 'link', {
+        'rel': 'stylesheet',
+        'type': 'text/css',
+        'href': url,
+      })
+    );
+  }
+
+  /**
+   * Creates the element and displays it on the page.
+   * @param {{
+   *   autoPromptType: (!AutoPromptType),
+   *   callback: (function()|undefined),
+   * }} options
+   */
+  create(options) {
+    if (
+      options.autoPromptType !== AutoPromptType.CONTRIBUTION &&
+      options.autoPromptType !== AutoPromptType.SUBSCRIPTION
+    ) {
+      return;
+    }
+
+    const theme = this.clientConfigManager_.getTheme();
+    const lang = this.clientConfigManager_.getLanguage();
+
+    // Create all the elements for the mini prompt.
+    /** @const {!Element} */
+    const miniPromptDiv = createElement(this.doc_.getWin().document, 'div', {});
+    miniPromptDiv.classList.add(`swg-mini-prompt-${theme}`);
+    miniPromptDiv.setAttribute('role', 'dialog');
+    if (lang) {
+      miniPromptDiv.setAttribute('lang', lang);
+    }
+
+    const titleContainerDiv = createElement(
+      this.doc_.getWin().document,
+      'div',
+      {}
+    );
+    titleContainerDiv.classList.add('swg-mini-prompt-title-container');
+    titleContainerDiv.setAttribute('role', 'button');
+
+    const iconDiv = createElement(this.doc_.getWin().document, 'div', {});
+    iconDiv.classList.add(`swg-mini-prompt-icon-${theme}`);
+
+    const titleTextH1 = createElement(this.doc_.getWin().document, 'h1', {});
+    titleTextH1.classList.add(`swg-mini-prompt-title-text-${theme}`);
+    if (options.autoPromptType === AutoPromptType.CONTRIBUTION) {
+      titleTextH1.textContent = msg(CONTRIBUTION_TITLE_LANG_MAP, lang) || '';
+    } else if (options.autoPromptType === AutoPromptType.SUBSCRIPTION) {
+      titleTextH1.textContent = msg(SUBSCRIPTION_TITLE_LANG_MAP, lang) || '';
+    }
+
+    const closeContainerDiv = createElement(
+      this.doc_.getWin().document,
+      'div',
+      {}
+    );
+    closeContainerDiv.classList.add('swg-mini-prompt-close-button-container');
+    closeContainerDiv.setAttribute('role', 'button');
+
+    const closeIconDiv = createElement(this.doc_.getWin().document, 'div', {});
+    closeIconDiv.classList.add(`swg-mini-prompt-close-icon-${theme}`);
+
+    // Now put it all together and insert it into the page.
+    closeContainerDiv.appendChild(closeIconDiv);
```

Comment:
I needed to leave the title and close container divs to be constructed in js so that I can reference them later for click/logging behavior, but I've moved their inner HTML to strings.